### PR TITLE
Workshop p.3 homework

### PR DESCRIPTION
**Pull request by vitalii-tkachuk**

### CODE REVIEW

File: internal/transport/grpc/server/users.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+func NewUsers(us handlers.UsersService, repo users.Repository) *Users {
+	return &Users{user: us, repo: repo}
+}
+
+func (u *Users) Create(ctx context.Context, in *pb2.CreateUserRequest) (*pb2.User, error) {
+	user, err := u.user.Create(ctx, in.GetName())
+	if err != nil {
+		return nil, err
```
Comment (mikenai) [inline internal/transport/grpc/server/users.go]: as well as HTTP, GRPC provides a response code. Try this https://pkg.go.dev/google.golang.org/grpc/status#Error

File: cmd/server/main.go [hunk 2]
```diff
@@ -83,11 +90,27 @@ func main() {
...
-		fmt.Println(s.ListenAndServe())
+		if err := httpServer.ListenAndServe(); err != http.ErrServerClosed {
+			log.Err(err).Msg("http server error received")
+			stop()
+		}
 	}()
 
-	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
-	defer stop()
+	grpcServer := grpc.NewServer()
```
Comment (mikenai) [inline cmd/server/main.go]: the GRPC server, as well as HTTP, accepts options - https://pkg.go.dev/google.golang.org/grpc#ServerOption
I wouldn't recommend any particular now, but it is worth looking at and remembering that it is there.

Comment (mikenai) [reply cmd/server/main.go]: also useful could be to have recovery middleware you could find such in this repo https://github.com/grpc-ecosystem/go-grpc-middleware

File: internal/transport/grpc/server/users.go (hunk 1 above)
Comment (vitalii-tkachuk) [reply internal/transport/grpc/server/users.go]: done

**Pull request merged**